libs/datamodel/core/src/ast/parser/parse_expression.rs

```diff
@@ -65,27 +61,17 @@ pub fn parse_arg_value(token: &Token) -> Expression {
 fn parse_string_literal(token: &Token) -> String {
     let current = token.first_relevant_child();
     match current.as_rule() {
-        Rule::string_content => unescape_string_literal(current.as_str()).into_owned(),
+        Rule::string_content => unescape_string_literal(current.as_str()),
         _ => unreachable!(
             "Encountered impossible string content during parsing: {:?}",
             current.tokens()
         ),
     }
 }
 
-fn unescape_string_literal(original: &str) -> Cow<'_, str> {
-    static STRING_LITERAL_UNESCAPE_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r#"\\(")"#).unwrap());
-    static STRING_LITERAL_BACKSLASHES_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r#"\\\\"#).unwrap());
-    static STRING_LITERAL_NEWLINE_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r#"\\n"#).unwrap());
-
-    match STRING_LITERAL_UNESCAPE_RE.replace_all(original, "\"") {
-        Cow::Owned(s) => match STRING_LITERAL_NEWLINE_RE.replace_all(&s, "\n") {
-            Cow::Owned(s) => STRING_LITERAL_BACKSLASHES_RE.replace_all(&s, "\\").into_owned().into(),
-            Cow::Borrowed(s) => STRING_LITERAL_BACKSLASHES_RE.replace_all(s, "\\").into_owned().into(),
-        },
-        Cow::Borrowed(s) => match STRING_LITERAL_NEWLINE_RE.replace_all(s, "\n") {
-            Cow::Owned(s) => STRING_LITERAL_BACKSLASHES_RE.replace_all(&s, "\\").into_owned().into(),
-            Cow::Borrowed(s) => STRING_LITERAL_BACKSLASHES_RE.replace_all(s, "\\").into_owned().into(),
-        },
-    }
+fn unescape_string_literal(original: &str) -> String {
+    original
+        .replace(r#"\""#, r#"""#)
+        .replace(r#"\\"#, r#"\"#)
+        .replace(r#"\\n"#, r#"\n"#)
```

Comment:
This is probably the piece that needs the closest review because I actually wrote logic here. @marcus given you're the original author can you validate this is correct?

Reply:
Unfortunately I am not the original author. I just moved this code into this file. But there are test cases for this. And they are failing on CI 🚨